UniWatch/Controllers/StudentController.cs

```diff
@@ -0,0 +1,191 @@
+                {
+                    Student = student,
+                    Enrolled = enrolledSet.Contains(student.Id)
+                });
+            }
+
+            return View("Enroll", viewModel);
+        }
+
+        // POST: /Student/Enroll?classId=1&studentId=1
+        /// <summary>
+        /// Enrolls a student into a class
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <param name="studentId">The id of the student to enroll</param>
+        [HttpPost]
+        [ActionName("Enroll")]
+        [ValidateAntiForgeryToken]
+        public ActionResult EnrollConfirmed(int classId, int studentId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            var student = _dataAccess.UserManager.GetStudentById(studentId);
+            if (@class == null || student == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            try
+            {
+                _dataAccess.ClassManager.EnrollStudent(classId, studentId);
+            }
+            catch (InvalidOperationException)
+            {
+                // Student is already enrolled in that class.
+                return new HttpStatusCodeResult(HttpStatusCode.Conflict);
+            }
+
+            return RedirectToAction("Index", new { classId = classId });
+        }
+
+        // GET: /Student/Unenroll?classId=1&studentId=1
+        /// <summary>
+        /// Returns the view to unenroll a student from a class 
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <param name="studentId">The id of the student</param>
+        [HttpGet]
+        public ActionResult UnEnroll(int classId, int studentId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            var student = _dataAccess.UserManager.GetStudentById(studentId);
+
+            if (@class == null || student == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            return View(new UnEnrollViewModel() { Class = @class, Student = student });
+        }
+
+        // POST: Unenroll
+        /// <summary>
+        /// Unenrolls a student from a class
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <param name="studentId">The id of the student to enroll</param>
+        [HttpPost]
+        [ActionName("UnEnroll")]
+        [ValidateAntiForgeryToken]
+        public ActionResult UnEnrollCofirmed(int classId, int studentId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            var student = _dataAccess.UserManager.GetStudentById(studentId);
+            if (@class == null || student == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
```

Comment:
Add a `string` inside the `HttpStatusCodeResult` explaining why the request is bad.

**i.e.** _No class with id [classId] or student with id [studentId]_